planning/behavior_path_start_planner_module/src/util.cpp

```diff
@@ -104,4 +105,35 @@ lanelet::ConstLanelets getPullOutLanes(
     /*forward_only_in_route*/ true);
 }
 
+double calcMinArcLengthDistanceFromEgoToObjects(
+  const tier4_autoware_utils::LinearRing2d & local_vehicle_footprint, const Pose & ego_pose,
+  const lanelet::ConstLanelets & lanelets, const PredictedObjects & static_objects)
+{
+  double min_distance = std::numeric_limits<double>::max();
+  const auto vehicle_footprint =
+    transformVector(local_vehicle_footprint, tier4_autoware_utils::pose2transform(ego_pose));
+  for (const auto & obj : static_objects.objects) {
+    const auto obj_polygon = tier4_autoware_utils::toPolygon2d(obj);
+    for (const auto & obj_outer_point : obj_polygon.outer()) {
+      const auto obj_pose_arc_length = getArcLengthForPoint(lanelets, obj_outer_point);
+      for (const auto & vehicle_corner_point : vehicle_footprint) {
+        const auto vehicle_pose_arc_length = getArcLengthForPoint(lanelets, vehicle_corner_point);
+        const double distance = std::abs(obj_pose_arc_length - vehicle_pose_arc_length);
+        min_distance = std::min(min_distance, distance);
+      }
+    }
+  }
```

Comment:
nits: 
this could be heavy.
I think there are better ways to make it faster, such as extracting the closest object, the closest point, etc.

Reply:
pseudo code
```
backward_objects = extraceBackwardObjects(lanelets, ego_pose, objectse); // extract objects whose arc length is less than ego's one
closest_backward_objects = getClosestObjects(backward_objects); // get object whose arc length is max
closet_object_corner_point = getClosestCornerPoint(closest_backward_objects); // get cornaner whose arc length is max
most_backward_ego_corner_point = getMostBackwardEgoCornerPoint(ego_pose, vehicle_footprint); // get corner whose arc length is min
distance = std::abs(most_backward_ego_corner_point - closet_object_corner_point); // get arc length between two points
```

Reply:
Thanks!
I will consider improving the efficiency

Reply:
In my current implementation, distance `a` is calculated but distance `b` should be calculated, correct?
If so, the method of extracting objects for back distance against back object should be changed. In this sketch, only object [2] should be extracted. 
![image](https://github.com/autowarefoundation/autoware.universe/assets/32741405/69cb1f1b-f238-4d14-b572-2c620078157f)

And for caring lateral collision implemented with the following implementetion. objects[0,1,2] should be extracted :thinking:   
```
      (utils::checkCollisionBetweenFootprintAndObjects(
        local_vehicle_footprint, *backed_pose, back_stop_objects_in_pull_out_lanes,
        parameters_->objects_collision_check_margins.back())))
```

Reply:
> I will consider improving the efficiency
thanks! but it's ok if it takes time to improve because the check target objects may be not so many 

> In my current implementation, distance a is calculated but distance b should be calculated, correct?
no, I think distance a should be calculated,

> And for caring lateral collision implemented with the following implementetion. objects[0,1,2] should be extracted 🤔
I agree

Reply:
I believe that some ingenuity is necessary in extracting objects to ensure `back_distance` in the lane coordinate system, especially in such unique situations. If we were to calculate `a` distance, I'm concerned that it might become impossible for backward driving.
![image](https://github.com/autowarefoundation/autoware.universe/assets/32741405/7010aa0f-bfe8-48bf-8168-c4e266b9247e)

red area represents `pull_out_lanes`

Reply:
yes, right. how about filter by lateral distance?

> backward_objects = extraceBackwardObjects(lanelets, ego_pose, objectse); // extract objects whose arc length is less than ego's one

pseudo
```
extraceBackwardObjects(const lanelet::ConstLanelets & lanelets, const geometry_msgs::msg::Pose & ego_pose, const PredictedObjects & objects, const double lateral_threshold){
  const double ego_arc_coords  = getArcCoordinates(lanelets, ego_pose.position).length;
  for (const auto & object : objects) {
    const double obj_arc_coods = getArcCoordinates(lanelets, object.pose.position);
    if(std::abs(ego_arc_coords.distance -obj_arc_coods.distance) > lateral_threshold){
      continue;
    }
    if (obj_arc_coods.length < ego_arc_coords.length) {
      backward_objects.push_back(object);
    }
  }
  return backward_objects;
}
```

Reply:
how about creating polygon along the `start_pose_candidates_path`?
https://github.com/autowarefoundation/autoware.universe/blob/1121f3b2a5a298ac474835522b97e84d9301dd9a/planning/behavior_path_start_planner_module/src/start_planner_module.cpp#L885

And extracts closest object overlaps with the polygon
![image](https://github.com/autowarefoundation/autoware.universe/assets/32741405/bc1b0bd0-7fc4-4b0e-8632-9bf3b04f6bd0)
a: `objects_collision_check_margins.back()`
b: `back_objects_collision_check_margin`

It's heavier but checking lateral distance of object pose is not considering the size of the object?

Reply:
looks good!
the attention area is  along the `start_pose_candidates_path` not nagative x-coords direction, right?

Reply:
which pattern did you mean?
![image](https://github.com/autowarefoundation/autoware.universe/assets/39142679/53e8b816-78d1-4db5-aa55-6c0fa7810fbc)

Reply:
and, better to check just side to ego?

![image](https://github.com/autowarefoundation/autoware.universe/assets/39142679/1d0d14f6-a172-424c-96b3-70cc2bb8db83)

Reply:
(sorry for my mistake, this pseudo code did not consider footprint)

https://github.com/autowarefoundation/autoware.universe/pull/6149#discussion_r1485610920

```
    const auto obj_arc_coods = getArcCoordinates(lanelets, object.pose.position);
    if(std::abs(ego_arc_coords.distance -obj_arc_coods.distance) > lateral_threshold){
      continue;
    }
```
not just check `object.pose.position` but need to check 4 footprint points by like `std::any`